### PR TITLE
Upgrading the `black` version

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes
I upgraded the version of `black` and reformatted the repository.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: examples/kserve_deployment/steps/pytorch_steps/mnist_handler.py [hunk 1]
```diff
@@ -19,7 +19,8 @@ class MNISTDigitClassifier(ImageClassifier):
     """
     MNISTDigitClassifier handler class. This handler extends class ImageClassifier from image_classifier.py, a
     default handler. This handler takes an image and returns the number in that image.
-    Here the post-processing method has been overridden while others are reused from parent class."""
+    Here the post-processing method has been overridden while others are reused from parent class.
```
Comment (strickvl) [inline examples/kserve_deployment/steps/pytorch_steps/mnist_handler.py:22]: Will `darglint` complain about this?

File: src/zenml/entrypoints/base_entrypoint_configuration.py [hunk 2]
```diff
@@ -157,6 +157,7 @@ def _parse_arguments(cls, arguments: List[str]) -> Dict[str, Any]:
         Raises:
             ValueError: If the arguments are not valid.
         """
+
```
Comment (strickvl) [inline src/zenml/entrypoints/base_entrypoint_configuration.py:160]: I think something will complain about this...

**Review by strickvl — APPROVED** [review]
Looks fine to me, but let's wait for final verdict from the CI... Feel like some of this would have triggered either `ruff` or `darglint` in the past?

File: examples/kserve_deployment/steps/pytorch_steps/mnist_handler.py (hunk 1 above)
Comment (bcdurak) [reply examples/kserve_deployment/steps/pytorch_steps/mnist_handler.py:22]: I ran `darglint` and `ruff` successfully before locally. Waiting for the checks to run first.

File: src/zenml/entrypoints/base_entrypoint_configuration.py (hunk 2 above)
Comment (bcdurak) [reply src/zenml/entrypoints/base_entrypoint_configuration.py:160]: Waiting on the checks myself.

**Review by fa9r — APPROVED** [review]
Reluctant approve since I think the new return type formatting changes suck. 

But if we decide to follow black's standards we might as well follow the latest version I guess.

**Pull request merged**